service/api_task.go

```diff
@@ -599,6 +599,19 @@ func assignNextAvailableTask(ctx context.Context, taskQueue *model.TaskQueue, di
 	})
 	stepStart = time.Now()
 
+	var amiUpdatedTime time.Time
+	if d.GetDefaultAMI() != currentHost.GetAMI() {
+		events, err := event.Find(event.AllLogCollection, event.DistroAMIModifiedForId(d.Id))
```

Comment:
Instead of doing a find all, should this just do a find one since it only cares about the first result anyways?

Reply:
I thought about that but we don't have a FindOne for events right now and I figured adding that should mean refactoring existing use cases. I can just add it to this though.